engine/src/main/java/org/terasology/network/BroadCastServer.java

```diff
@@ -45,26 +44,29 @@ public void startBroadcast() {
             Executors.newSingleThreadExecutor().execute(new Runnable() {
                 @Override
                 public void run() {
-                    while (true) {
+                    while (!Thread.currentThread().isInterrupted()) {
```

Comment:
@DarkWeird Added the thread interruption condition, so that we don't wait continously